dotCMS/src/test/java/com/dotmarketing/portlets/workflows/actionlet/SaveContentAsDraftActionletTest.java

```diff
@@ -0,0 +1,149 @@
+package com.dotmarketing.portlets.workflows.actionlet;
+
+import com.dotmarketing.beans.Permission;
+import com.dotmarketing.business.PermissionAPI;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.exception.DotSecurityException;
+import com.dotmarketing.portlets.categories.business.CategoryAPI;
+import com.dotmarketing.portlets.categories.model.Category;
+import com.dotmarketing.portlets.contentlet.business.ContentletAPI;
+import com.dotmarketing.portlets.contentlet.model.Contentlet;
+import com.dotmarketing.portlets.contentlet.model.ContentletDependencies;
+import com.dotmarketing.portlets.structure.model.ContentletRelationships;
+import com.dotmarketing.portlets.workflows.model.WorkflowActionClassParameter;
+import com.dotmarketing.portlets.workflows.model.WorkflowProcessor;
+import com.liferay.portal.model.User;
+import com.tngtech.java.junit.dataprovider.DataProvider;
+import com.tngtech.java.junit.dataprovider.DataProviderRunner;
+import com.tngtech.java.junit.dataprovider.UseDataProvider;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+import java.util.List;
+import java.util.Map;
+
+import static com.dotcms.util.CollectionsUtils.map;
+import static org.mockito.Mockito.*;
+
+@RunWith(DataProviderRunner.class)
+public class SaveContentAsDraftActionletTest {
+
+    private SaveContentAsDraftActionlet saveContentAsDraftActionlet;
+    private ContentletAPI contentletAPI;
+    private CategoryAPI categoryAPI;
+    private PermissionAPI permissionAPI;
+
+    @Before
+    public void init(){
+        contentletAPI = mock(ContentletAPI.class);
+        categoryAPI = mock(CategoryAPI.class);
+        permissionAPI = mock(PermissionAPI.class);
+
+        saveContentAsDraftActionlet = new SaveContentAsDraftActionlet(contentletAPI, categoryAPI, permissionAPI);
+    }
+
+    private static class TestCase {
+        final private boolean respectFrontendRoles;
+
+        private TestCase(boolean respectFrontendRoles) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Parameter 'respectFrontendRoles' is not assigned and could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5337055)